windows/devices/uwb/UwbConnector.cxx

```diff
@@ -875,48 +895,95 @@ UwbConnector::NotificationListenerStop()
     m_notificationThread.request_stop();
 }
 
-::uwb::RegisteredCallbackToken*
+std::weak_ptr<::uwb::RegisteredCallbackToken>
 UwbConnector::RegisterDeviceEventCallbacks(std::weak_ptr<::uwb::UwbRegisteredDeviceEventCallbacks> callbacks)
 {
     std::lock_guard eventCallbacksLockExclusive{ m_eventCallbacksGate };
     bool isFirstCallback = not CallbacksPresent();
-    m_deviceEventCallbacks.push_back(std::move(callbacks));
+    auto token = InsertDeviceEventCallback(callbacks);
     if (isFirstCallback) {
         NotificationListenerStart();
     }
-    return nullptr;
+    return token;
 }
 
-::uwb::RegisteredCallbackToken*
+std::weak_ptr<::uwb::RegisteredCallbackToken>
 UwbConnector::RegisterSessionEventCallbacks(uint32_t sessionId, std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbacks> callbacks)
 {
     std::lock_guard eventCallbacksLockExclusive{ m_eventCallbacksGate };
     bool isFirstCallback = not CallbacksPresent();
-    // Obtain the map node to the existing vector of callbacks, if present.
-    auto node = m_sessionEventCallbacks.extract(sessionId);
-    if (!node.empty()) {
-        auto& sessionEventCallbacks = node.mapped();
-        sessionEventCallbacks.push_back(std::move(callbacks));
-        m_sessionEventCallbacks.insert(std::move(node));
-    } else {
-        m_sessionEventCallbacks.insert({ sessionId, { std::move(callbacks) } });
-    }
-
+    auto token = InsertSessionEventCallback(sessionId, callbacks);
     if (isFirstCallback) {
         NotificationListenerStart();
     }
-    return nullptr;
+    return token;
 }
 
 bool
 UwbConnector::CallbacksPresent()
 {
-    return (not m_deviceEventCallbacks.empty()) or (not m_sessionEventCallbacks.empty());
+    return (not m_sessionEventCallbacks.empty()) or (not m_deviceEventCallbacks.empty());
 }
 
 void
-UwbConnector::DeregisterEventCallback(::uwb::RegisteredCallbackToken* token)
+UwbConnector::DeregisterEventCallback(std::weak_ptr<::uwb::RegisteredCallbackToken> token)
 {
-    // TODO implement
+    auto tokenShared = token.lock();
+    if (not tokenShared) {
+        return;
+    }
     std::lock_guard eventCallbacksLockExclusive{ m_eventCallbacksGate };
+
+    auto deviceCallback = std::dynamic_pointer_cast<::uwb::RegisteredDeviceCallbackToken>(tokenShared);

```

Comment:
nit: It might be worth reversing the order of the checks to do the `::uwb::RegisteredSessionCallbackToken` type first since we would expect the callbacks for sessions to come and go regularly, but the ones for device to remain fairly static. Basically, put the likely branch condition first.